src/client/shadows/dynamicshadowsrender.cpp

```diff
@@ -108,11 +100,17 @@ void ShadowRenderer::preInit(IWritableShaderSource *shsrc)
 {
 	if (g_settings->getBool("enable_dynamic_shadows")) {
 		shsrc->addShaderUniformSetterFactory(new ShadowUniformSetterFactory());
+		shsrc->addShaderUniformSetterFactory(new ShadowScreenQuadUniformSetterFactory());
 	}
 }
 
 void ShadowRenderer::initialize()
 {
+	if (g_settings->getBool("enable_dynamic_shadows")) {
+		auto* shdsrc = m_client->getShaderSource();
+		shdsrc->addShaderUniformSetterFactory(new ShadowDepthUniformSetterFactory(this));
```

Comment:
Cannot be done in `ShadowRenderer::preInit` because it needs a pointer to "this", and preInit is a static method called before this class is initialize... This is a part of the *UniformSetterFactory* pattern I don't like TBH.